Upgraded some libraries for 6.3

MarekB01 opened the pull request:
Upgraded Spring boot to 2.7.3 and Hibernate to 5.6.11

Link to QA https://github.com/BroadleafCommerce/QA/issues/4786

MarekB01 committed:
BroadleafCommerce/QA-4786_library_upgrades
Upgraded Spring boot ro 2.7.3 and Hibernate to 5.6.11

MarekB01 committed:
BroadleafCommerce/QA-4786_library_upgrades
Marked Country.java and CountryImpl.java as deprecated, changed application context for bl-profile-applicationContext-entity.xml

oleksiimiroshnyk commented on common/pom.xml:
```diff
@@ -452,5 +452,9 @@
             <groupId>com.googlecode.htmlcompressor</groupId>
             <artifactId>htmlcompressor</artifactId>
         </dependency>
+        <dependency>
```
Why did you add new dependency?

oleksiimiroshnyk commented on admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/dao/provider/metadata/MapFieldsFieldMetadataProvider.java:
```diff
@@ -171,7 +170,6 @@ public MetadataProviderResponse addMetadataFromFieldType(AddMetadataFromFieldTyp
                         ((BasicFieldMetadata) entry.getValue()).getExplicitFieldType(),
                         myType.getReturnedClass(), addMetadataFromFieldTypeRequest.getDynamicEntityDao()), metadata);
 
-                TypeConfigurationRegistry.INSTANCE.deregisterTypeConfiguration(typeConfiguration);
```
is this class missing ? or why did you delete it ?

MarekB01 replied:
It is missing, because Hibernate was upgraded to 5.6.11

MarekB01 committed:
BroadleafCommerce/QA-4786_library_upgrades
Marked Country.java and CountryImpl.java as deprecated, changed application context for bl-profile-applicationContext-entity.xml

riteshadhikari17 commented on core/broadleaf-profile/src/main/java/org/broadleafcommerce/profile/core/domain/Country.java:
```diff
@@ -33,6 +33,7 @@
  *
  * @author Elbert Bautista (elbertbautista)
  */
+@Deprecated
```
@oleksiimiroshnyk Shouldn't we remove this instead of deprecating? I thought [this ticket](https://github.com/BroadleafCommerce/QA/issues/4509) was for removing. Its 6.3 so its better remove it and in Migration notes mention to use `org.broadleafcommerce.common.i18n.domain.ISOCountryImpl`

riteshadhikari17 replied:
@MarekB01 I wasn't talking about removing deprecation tag. I was asking if we should remove these `Country` and `CountryImpl` classes. 
BTW in future, remember to mention the class that should be used along with deprecation tag.

oleksiimiroshnyk replied:
@riteshadhikari17 after investigation we decided to not remove CountryImpl, looks like its main purpose is to be "supported country dictionary" while IsoCountry is a global dictionary.
So for now lets leave as it is. 

If we want we can create a separate task to get rid of CountryImpl completely and move it is goal to, let's say, properties or somewhere else.

MarekB01 committed:
BroadleafCommerce/QA-4786_library_upgrades

oleksiimiroshnyk commented on core/broadleaf-profile/src/main/resources/bl-profile-applicationContext-entity.xml:
```diff
@@ -23,9 +23,9 @@
-        <property name="country">
-            <bean class="org.broadleafcommerce.profile.core.domain.CountryImpl">
-                <property name="abbreviation" value="US"/>
+        <property name="isoCountryAlpha2">
```
I think this can be reverter ? as we will not delete CountryImpl

MarekB01 replied:
It cannot see the property with name "country" I think the field Country was removed with State in AddressImpl. I think in this case it would be better to leave the changed code
